### PR TITLE
Back/cache workflow

### PR DESCRIPTION
**Pull request by cristian-tamblay**
# New cache for github workflow
This cache reduces 50% time on github's python installation

## Changes

- Implementented a python env cache, that is bypassed only when modifying setup.py, requirements or requirements-dev

### CODE REVIEW

**Comment by cristian-tamblay** [comment]
There is a problem with extras_require and pip install -e, if anyone has an idea how to fix it.
image

This improved pip install time from 6 minutes to 3minutes when using cache

File: .github/workflows/build-test.yaml [hunk 1]
```diff
@@ -32,9 +32,14 @@ jobs:
...
         with:
           python-version: '3.10'
-          cache: 'pip'
-      - run: pip install -r requirements.txt
-      - run: pip install -r requirements-dev.txt
+        
+      - uses: actions/cache@v3
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements-dev.txt') }}
```
Comment (pbadillatorrealba) [inline .github/workflows/build-test.yaml:39]: If I understand correctly, this generates a directory for each different version of setup.py, requirements.txt and requirements-dev.txt?


What happens with outdated environments, are they deleted after a while?

Comment (cristian-tamblay) [reply .github/workflows/build-test.yaml:39]: image

Comment (pbadillatorrealba) [reply .github/workflows/build-test.yaml:39]: Nice!

**Review by pbadillatorrealba — APPROVED** [review]

**Pull request merged**